### PR TITLE
Replace Version with semver lib

### PR DESCRIPTION
**Pull request by lightglitch**
First Step to #32

### CODE REVIEW

**Comment by decebals** [comment]
My philosophy (see my projects, [pippo](https://github.com/decebals/pippo) for example) is to keep the number of dependencies as low as possible, because people/users want this. I add a new dependency only if that library brings value in project.
I think that Version is a decent, uncompleted implementation of semver.  If you see some problems in this class please tell me.

**Comment by lightglitch** [comment]
I understand that and the value that the lib brings is the support for semver ranges (https://github.com/zafarkhaja/jsemver#semver-expressions-api-ranges) that I'm going to need for compatibility check of plugins.

**Comment by decebals** [comment]
Thanks for your explanation. Can you comment on how you see # 32 implementation to have a common understanding?

**Comment by lightglitch** [comment]
I still thinking about it because I want to use the process information of the graph to do the validations.

But as a big picture I just going to put the plugin with the disable state if some of his dependencies are disabled. This with some error reporting information is enough for me.

**Comment by decebals** [comment]
@gitblit I think that it's a good idea to have range validation on plugin version and I think that the `jsemver` library is a solid choice (no dependency, active development - I don't wish to implement a complex semver in pf4j). Is it ok for you this change?

**Comment by gitblit** [comment]
It's ok with me.

File: pf4j/src/main/java/ro/fortsoft/pf4j/PluginDependency.java [hunk 1]
```diff
@@ -12,38 +12,38 @@
...
-
-	@Override
-	public String toString() {
-		return "PluginDependency [pluginId=" + pluginId + ", pluginVersion=" + pluginVersion + "]";
-	}
+    private String pluginId;
+    private String pluginVersionSupport;
+
+    public PluginDependency(String dependency) {
+        int index = dependency.indexOf('@');
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginDependency.java]: why `@` and not `:`?

Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginDependency.java]: For me the sintax plugin-at-version makes more sense but I saw that you were using : and forgot to change it.

Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginDependency.java]: No problem, we can keep `@`

**Pull request merged**